Cesium.Runtime/StdIoFunctions.cs

```diff
@@ -668,15 +667,62 @@ public static int Remove(byte* pathname)
         return 0;
     }
 
-    private static unsafe void* GetLastStream()
+    internal static StreamHandle? GetStream(IntPtr handle)
     {
-        return (void*)(IntPtr)(Handles.Count - 1);
+        var handleValue = handle.ToInt64();
+        if (handleValue is StdIn or StdOut or StdErr)
+        {
+            return Handles[(int)handleValue];
+        }
+
+        try
+        {
+            lock (_locker)
+            {
+                var gchAddr = Marshal.ReadIntPtr(handle);
+                var gch = GCHandle.FromIntPtr(gchAddr);
+
+                return gch.Target as StreamHandle;
+            }
+        }
+        catch (AccessViolationException)
+        {
+            return null;
+        }
```

Comment:
There's no point in trying to catch this. The program should fail if an exception is thrown here.